### PR TITLE
LPS-113192 update values in other languageIds when it changes in default languageId

### PR DESCRIPTION
**Pull request by marcosapmf**
Related issue: https://issues.liferay.com/browse/LPS-113192

### CODE REVIEW

**Comment by natocesarrego** [comment]
ci:test:forms

**Comment by natocesarrego** [comment]
Hey @brunofarache,

Can you please review this pull? It's a pull exclusive related to React/JavaScript and I understand that you're better prepared to perform this review. I already started Forms test suite to verify if this proposed fix introduces failures.

Thanks in advance.

**Comment by brunofarache** [comment]
Just started reviewing :)

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/Options/Options.es.js [hunk 1]
```diff
@@ -66,6 +67,23 @@ const Option = React.forwardRef(
...
+				editingLanguageId,
+				options,
+				option
+			),
+			id: random(),
+		})),
+		defaultLanguageId === editingLanguageId
+			? {...defaultOption, generateKeyword: true, id: random()}
+			: false,
+	].filter(Boolean);
```
Comment (brunofarache) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/Options/Options.es.js:85]: @marcosapmf o que seria `.filter(Boolean)`? tem alguma maneira de ser mais explicíto no que queres fazer?

por exemplo `.filter(field -> field)`?

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/Options/Options.es.js [hunk 2]
```diff
@@ -122,11 +146,43 @@ const Options = ({
...
+			});
+
+			return _normalizedValue;
 		}
 
 		return {...normalizedValue, [editingLanguageId]: _fields};
 	};
 
+	const synchroniseValue = (fields, languageId) => {
+		const _value = [];
```
Comment (brunofarache) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/Options/Options.es.js:170]: não seria `values` o nome da variavel nao? já que é um array....

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/Options/Options.es.js [hunk 3]
```diff
@@ -122,11 +146,43 @@ const Options = ({
...
+
+		for (const index in fields) {
```
Comment (brunofarache) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/Options/Options.es.js:172]: equivalente a `fields.forEach(...)`?

**Comment by natocesarrego** [comment]
Pull request forwarded to https://github.com/brunofarache/liferay-portal/pull/303. <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Pull request closed without merging**